plugins/module_utils/network/iosxr/facts/vrf/vrf.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+"""
+The iosxr vrf fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+
+from copy import deepcopy
+
+from ansible.module_utils.six import iteritems
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import utils
+
+from ansible_collections.cisco.iosxr.plugins.module_utils.network.iosxr.argspec.vrf.vrf import (
+    VrfArgs,
+)
+from ansible_collections.cisco.iosxr.plugins.module_utils.network.iosxr.rm_templates.vrf import (
+    VrfTemplate,
+)
+from ansible_collections.cisco.iosxr.plugins.module_utils.network.iosxr.utils.utils import (
+    flatten_config,
+)
+
+
+class VrfFacts(object):
+    """The iosxr vrf facts class"""
+
+    def __init__(self, module, subspec="config", options="options"):
```

Comment:
This method does not seem to be generated using content_builder. Please have a look and update.

https://github.com/ansible-community/ansible.content_builder/blob/main/roles/module_network_cli/templates/module_utils/network_os/facts/resource/resource.py.j2#L34-L37

Reply:
These assignments and logic is redundant.